### ISSUE TITLE
溜まったにっぽうの扱いを考える

### ISSUE THREAD

**Issue by macha3162** [chibidata/dekimashita#48]
日報は現在どんどんと溜まっていきます。
セキュリティー的にリスクになるので、
まとめメールを送った段階で消すとか扱いを考えたいです。

ちなみに、外部のメールサービスであるかPostMarkにも、
メールが参照できる形で残っているので、こちらに関しても扱いを考える。

**Comment by macha3162** [comment]
現状の追記
ガイドには、まとめが送られると削除されますと書かれている。
アカウントを退会した時には、全データ消している。

**Comment by yoshi-takeda** [comment]
**構成案(細かいアーキテクチャは任せる)**

前提
---
メール送受信はSESでやる
SESを前提にするとAWSのセキュリティに乗っかれるのでそれが前提で良いと判断
slackへの判定例からヤフー側でもAWSのセキュリティ自体は高いと判定されている

メール送信(質問、初回、まとめ、時間外)
---
・SESから送信する
・送信事実(対象、送信日時、送信数)だけを残す

メール受信
---
・SESで受信する（※ここのパース処理がちょっと懸念あり。手間が結構増えるかも)
・受信事実(対象、受信日時)と、パース結果を「一時的に」保存する（※本文じゃなくてパース結果なので注意）
・まとめメール送信が終わったら一時的に保存したパース結果のデータは消す

**Comment by yoshi-takeda** [comment]
そういう前提で、具体的に残すべき項目は画面と照らしあわせて精査します

**Comment by yeesamson** [comment]
データを保持するリスクとコストがありますので、そこはできれば避けたい。idonethisのような見返すことはメールに任せましょう。

**Comment by macha3162** [comment]
sesを調査いています。
結論、構築に２日はかかるのではなかろうか。

sesはメールを受信する仕組みはあるのですが、アプリケーションにPOSTする方法がないので自前で構築する必要があります。 試行錯誤の結果、受信したメールをS3に一度保存、保存をフックにlambdaを動かして、アプリケーションにPOSTする。　この構成しかなさそうです。
S3に一時保存したメールは、処理が成功したタイミングで削除します。

Herokuでお手軽に構築できるサービスを1個作るイメージなので、時間はかかります。

**Comment by yeesamson** [comment]
はい。SESを使うメリットと、いまのままのやり方との違いはメールを送る際のセキュリティが高まるのと、一時保存されている情報のセキュリティも高められるイメージでしょうか。

**Comment by macha3162** [comment]
はい。　高まります。

自前で構築した、システムであること。
データを消すタイミングを完全にこちらから制御できるためです。

**Comment by yeesamson** [comment]
運用コストとスケーラビリティはどんな感じですか。たとえは日報が届く人が1,000人、10,000人の場合は月額いくらを想定してますか。

**Comment by macha3162** [comment]
AWSの料金
月に60,000通までは、無料で使えます。
ということで、1000人の場合は無料でまかなえます。
無料を無視して、1日に15,000通通信がある場合、コストは月額5000円ほどです。

これは、HerokuのParseサービスに比べると格安です。
月の送信数15,000x30=450,000通

PostMarkだと 500,000通 $500/月
Sendgridだと 700,000通 $400/月

**Comment by yeesamson** [comment]
ありがとうございます。
このタスクは増田さんもやるべきという判断であれば、２日かけてやりましょう。

**Comment by yoshi-takeda** [comment]
やります

**Comment by macha3162** [comment]
AWSに環境ができました。
http://yoku.dekimashita.com/

TODO、ドメイン変更(SSL証明書はLBに設定済みなので、SSLは有効になる)
懸念点、メーラーによって、Textメールの場所が異なると、Parseに失敗するかもしれない。

**Comment by yoshi-takeda** [comment]
竹田の方で動作確認します！
